drivers/adc/adc_stm32.c

```diff
@@ -407,15 +407,17 @@ static void adc_stm32_oversampling(ADC_TypeDef *adc, uint8_t ratio, uint32_t shi
  */
 static int adc_stm32_enable(ADC_TypeDef *adc)
 {
-	LL_ADC_Enable(adc);
-
 #if defined(CONFIG_SOC_SERIES_STM32L4X) || \
 	defined(CONFIG_SOC_SERIES_STM32L5X) || \
 	defined(CONFIG_SOC_SERIES_STM32WBX) || \
 	defined(CONFIG_SOC_SERIES_STM32G0X) || \
 	defined(CONFIG_SOC_SERIES_STM32G4X) || \
 	defined(CONFIG_SOC_SERIES_STM32H7X) || \
 	defined(CONFIG_SOC_SERIES_STM32WLX)
+
+	LL_ADC_ClearFlag_ADRDY(adc);
```

Comment:
```
	LL_ADC_ClearFlag_ADRDY(adc);
	if (LL_ADC_IsEnabled(adc) == 1UL) {
		return 0;
	}
	LL_ADC_Enable(adc);
```

Reply:
I took your suggestion but put it before LL_ADC_ClearFlag_ADRDY as well because there is no point in clearing the flag if we're not gonna await it.

I tested this before and after on a nucleo_wb55rg which I just got in the mail. :) It also had the deadlock issue.